yard-mode.el

```diff
@@ -191,17 +191,26 @@ The format is suitable for `font-lock-add-keywords' and `font-lock-remove-keywor
     (let ((tag (yard-tag-at-point)))
       (when tag (yard-tag-syntax tag)))))
 
-(defun yard-turn-on ()
-  "Turn on yard-mode."
-  (font-lock-add-keywords nil (yard-font-lock-keywords))
-  (when yard-use-eldoc
-    (set (make-local-variable 'eldoc-documentation-function) 'yard-eldoc-message)))
-
-(defun yard-turn-off ()
-  "Turn off yard-mode."
-  (font-lock-remove-keywords nil (yard-font-lock-keywords))
-  (when yard-use-eldoc
-    (set (make-local-variable 'eldoc-documentation-function) nil)))
+(eval-and-compile 
```

Comment:
If you want to remove the conditional branch from `.elc` file , just use `(eval-when-compile (fboundp 'remove-function))`.  However, the runtime cost saved by this change is negligible.

[Knuth's advice in 1979](https://en.wikiquote.org/wiki/Donald_Knuth#Computer_Programming_as_an_Art_(1974)):
> The real problem is that programmers have spent far too much time worrying about efficiency in the wrong places and at the wrong times; **premature optimization is the root of all evil (or at least most of it) in programming**.

Optimization advice is also summarized in [Rob Pike's 5 Rules of Programming](http://users.ece.utexas.edu/~adnan/pike.html).

Reply:
I'm sorry, I misunderstood the purpose of this change. However, `eval-and-compile` still has no effect. To deal with test execution and byte compiler warnings, just branching with `(fboundp 'remove-function)` is sufficient.

Reply:
@zonuexe well that was the original change, but that isn't sufficient to avoid the eager macroexpansion in emacs 24.3/24.4.  That is why the `eval-and-compile` is there -- not as an optimization